### PR TITLE
RDM-9555: Definition Store Changes for Conditional Event Post State

### PR DESCRIPTION
**Pull request by kiran-yenigala-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDM-9555

### Change description ###

Process Conditional Post State events

**Does this PR introduce a breaking change?** (check one with "x")

```
[ X] Yes
[ ] No
```

### CODE REVIEW

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventPostStateParserTest.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+        caseType = new CaseTypeEntity();
+        caseType.setReference(CASE_TYPE_UNDER_TEST);
+
+        definitionSheets.put(SheetName.CASE_EVENT.getName(), definitionSheet);
+        caseEventToFieldsSheet = new DefinitionSheet();
+        definitionSheets.put(SheetName.CASE_EVENT_TO_FIELDS.getName(), caseEventToFieldsSheet);
+        when(parseContext.getStateForCaseType(anyString(), anyString())).thenReturn(new StateEntity());
+
+        ch.qos.logback.classic.Logger root = (ch.qos.logback.classic.Logger) LoggerFactory.getLogger(ch.qos.logback.classic.Logger.ROOT_LOGGER_NAME);
```
Comment (rafalkalita) [inline excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventPostStateParserTest.java:65]: Do we need this and the mockAppender?

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java [hunk 2]
```diff
@@ -592,7 +591,7 @@ void testMapEventEntity() {
                     parameters
                 );
                 assertEquals(parameters.getPreStateExpectation(), caseEvent.getPreStates());
-                assertEquals(parameters.getPostStateExpectation(), caseEvent.getPostState());
+//                assertEquals(parameters.getPostStateExpectation(), caseEvent.getPostState());
```
Comment (rafalkalita) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java:594]: Please fix or remove.

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java [hunk 3]
```diff
@@ -624,7 +623,7 @@ void testMapEmptyEventEntity() {
 
             assertEquals(1, caseEvent.getPreStates().size());
             assertThat(caseEvent.getPreStates(), hasItems("*"));
-            assertEquals("*", caseEvent.getPostState());
+//            assertEquals("*", caseEvent.getPostState());
```
Comment (rafalkalita) [inline domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java:626]: Please fix or remove.

File: excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventPostStateParserTest.java (hunk 1 above)
Comment (kiran-yenigala-hmcts) [reply excel-importer/src/test/java/uk/gov/hmcts/ccd/definition/store/excel/parser/EventPostStateParserTest.java:65]: Removed un-used root and mockAppender

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java (hunk 3 above)
Comment (kiran-yenigala-hmcts) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java:626]: fixed it...

File: domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java (hunk 2 above)
Comment (kiran-yenigala-hmcts) [reply domain/src/test/java/uk/gov/hmcts/ccd/definition/store/domain/service/EntityToResponseDTOMapperTest.java]: fixed it..

**Review by rafalkalita — APPROVED** [review]

**Pull request merged**